onnxruntime/core/providers/nuphar/compiler/x86/op_ir_creator/nn/pool_ops.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/nuphar/compiler/x86/op_ir_creator/all_ops.h"
+
+#include "core/codegen/passes/utils/ort_tvm_utils.h"
+#include "core/framework/op_kernel_info.h"
+#include "core/providers/cpu/nn/pool_attributes.h"
+#include "core/providers/nuphar/mti_x86/nn/pool_ops.h"
+
+namespace onnxruntime {
+namespace nuphar {
+
+static tvm::Array<tvm::Expr> GetOutputShape(const Node& node,
+                                            const PoolAttributes& pool_attrs,
+                                            tvm_codegen::CodeGenContext& ctx_codegen) {
+  const NodeArg* input = node.InputDefs()[0];
+  ORT_ENFORCE(input);
+  const ONNX_NAMESPACE::TensorShapeProto* shape_proto = input->Shape();
+  size_t num_input_dims = shape_proto->dim_size();
+  ORT_ENFORCE(num_input_dims >= 2);
+
+  tvm::Array<tvm::Expr> output_shape;
+  // batch dimenion
+  output_shape.push_back(ShapeDimToTvmDim(shape_proto->dim(0), ctx_codegen));
+  // output channel
+  output_shape.push_back(ShapeDimToTvmDim(shape_proto->dim(1), ctx_codegen));
+
+  if (pool_attrs.global_pooling) {
+    for (size_t dim = 2; dim < num_input_dims; dim++) {
```

Comment:
>2 [](start = 22, length = 1)

could be more general to use rank of kernel_shape, instead of hardcoded for 2D only.

Reply:
It's not hardcoded for 2D. Note that the first two dims are batch dim and channel dim, so ```num_input_dims - 2``` is the same as kernel_shape.size().

Reply:
You are right, please try use kernel_shape.size() if possible and avoid magic numbers 2 here.

---
In reply to: [327992038](https://github.com/microsoft/onnxruntime/pull/1914#discussion_r327992038) [](ancestors = 327992038)